### PR TITLE
Setup cron job to build nightly docker image

### PR DESCRIPTION
**Pull request by gayyappan**

### CODE REVIEW

File: .github/workflows/docker-nightly-build.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Docker Image Nightly Build CI
+
+on:
+  push:
+    branches: [ cron_build ]
```
Comment (mkindahl) [inline .github/workflows/docker-nightly-build.yml:5]: No need to keep this after the experiment is done.

File: .github/workflows/docker-nightly-build.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+env:
+  ORG: timescaledev
+
+jobs:
+
+  # Build ubuntu TimescaleDB images for both TSL and OSS code.
+  timescaledb:
+
+    name: PG${{ matrix.pg }}${{ matrix.oss }}
```
Comment (mkindahl) [inline .github/workflows/docker-nightly-build.yml:19]: Missing an `oss` in the matrix?

File: .github/workflows/docker-nightly-build.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      uses: docker/setup-buildx-action@v1
+
+    - name: Linux available buildx platforms 
+      run: echo ${{ steps.buildx.outputs.platforms }}
+
+    - name: Login to DockerHub Registry
+      run: echo ${{ secrets.DOCKERHUB_TOKEN }} | docker login -u ${{ secrets.DOCKERHUB_USER }} --password-stdin
+
+    - name: Build and push nightly Docker image for TimescaleDB
+      run: make nightly ORG=$ORG PG_VER=pg${{ matrix.pg }}
```
Comment (mkindahl) [inline .github/workflows/docker-nightly-build.yml:40]: Don't you need to handle both OSS and TSL version when you build them?

**Review by mkindahl — COMMENTED** [review]
Using Makefiles in this manner is kind of hackish, but it works. Might be better to just have a script for it, but that is a later problem.

Do you want to build the OSS version for each package as well?

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -5,7 +5,7 @@ on:
     branches: [ master, dev-build ]
 
 env:
-  ORG: timescale # timescaledev
+  ORG: timescaledev #timescale
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply .github/workflows/docker-image.yml:9]: Won't the merge of this PR to master trigger a docker image build? Is that an incorrect inference?

File: .github/workflows/docker-nightly-build.yml (hunk 2 above)
Comment (gayyappan) [reply .github/workflows/docker-nightly-build.yml:19]: will remove oss. Don't need oss nightly builds.

File: .github/workflows/docker-nightly-build.yml (hunk 1 above)
Comment (gayyappan) [reply .github/workflows/docker-nightly-build.yml:5]: Isn't it still useful to have this to debug any job failures? That's the reason I kept it. No strong opinion either way.

File: .github/workflows/docker-nightly-build.yml (hunk 3 above)
Comment (gayyappan) [reply .github/workflows/docker-nightly-build.yml:40]: removed oss.

**Review by cevian — COMMENTED** [review]
LGTM. thank you.

File: .github/workflows/docker-image.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkindahl) [reply .github/workflows/docker-image.yml:9]: It will indeed. We need to remember to change this back for the next release then.

**Review by mkindahl — APPROVED** [review]
LGTM

**Review by svenklemm — COMMENTED** [review]
For debugging it would be really nice to have the latest-tagging only happen for cron-builds

File: .github/workflows/docker-image.yml [hunk 5]
```diff
@@ -5,7 +5,7 @@ on:
     branches: [ master, dev-build ]
```
Comment (svenklemm) [inline .github/workflows/docker-image.yml:5]: i suggest removing master from this and add release_docker here so random commits on master don't overwrite the latest tagged already released docker images

Comment (gayyappan) [reply .github/workflows/docker-image.yml]: removed master and added release_docker.

**Review by svenklemm — APPROVED** [review]

**Pull request merged**